provider/ansible.py

```diff
@@ -170,7 +170,13 @@ def distro_install(packages="ansible"):
     package_list = params.get_list("package_list", 'sshpass')
     try:
         path.find_command('ansible-playbook')
-    except path.CmdNotFoundError:
+        # If ansible debug plugin is required we need to enforce full
+        # ansible version installation instead of ansible-core
+        if 'debug' in params.objects('ansible_callback_plugin'):
```

Comment:
Maybe we can check all callback plugins, not only for `debug`.

Reply:
I see your point. What I am thinking about is: Are the other plugins really a must for execution of ansible-related test cases? If so, let's add them. What do you think?

Reply:
I will try to execute ansible test case with `ansible-core` + `ansible.posix` (via `ansible-galaxy collection install ansible.posix`) to see if this is sufficient combination to run our automation. Please correct me, if I am wrong but  - according to https://docs.ansible.com/ansible/latest/collections/ansible/posix/debug_callback.html - it should be only possible to install all the `ansible.posix` plugins at once, not to limit it to debug plugin only.

Reply:
IIUC, we can only use 1 callback plugin when executing ansible command, https://docs.ansible.com/ansible/latest/plugins/callback.html#setting-a-callback-plugin-for-ansible-playbook

So this change is wrong. @yanan-fu I saw you add this suggestion, please correct me if I mistake.

Reply:
> IIUC, we can only use 1 callback plugin when executing ansible command, https://docs.ansible.com/ansible/latest/plugins/callback.html#setting-a-callback-plugin-for-ansible-playbook
> 
> So this change is wrong. @yanan-fu I saw you add this suggestion, please correct me if I mistake.

Not exactly,  as my understanding, we can use multiple callback plugins, but for the current code, the param `ansible_callback_plugin` only be used for `stdout`.
https://github.com/autotest/tp-qemu/blob/d0731d9775fde994ec37dddbbd5c1ab6dd4aab2a/provider/ansible.py#LL63C18-L63C18
That is why i mentioned: https://github.com/autotest/tp-qemu/pull/3688#discussion_r1180048556

But right, it is a problem that we handle it as ansiable callback plugin here but use it as stdout callback only.

As our discussion offline, considering we only use stdout callback,  let's update the param name from `ansible_callback_plugin` to `ansible_stdout_callback`.

Reply:
@yanan-fu @PaulYuuu It is possible to change the naming (I agree the new name it is more clear), but it would also affect number of test cases (7 of them) and other users + automation we may be not aware of:
```
ansible_test.py
ansible_with_responsive_migration.py
ansible_test.cfg
nested_block_resize.cfg
nested_vsock_con_sockets.cfg
nested_block_resize.py
nested_vsock_con_sockets.py
```
In the meantime I considered another solution. Goal of this PR is to ansure we have full ansible version when it is necessary. Also the problem applies only to situation if `python_install` is in place.

What do you think about following?
```
    ansible_install_policy = params['ansible_install_policy']
    package_list = params.get_list("package_list", 'sshpass')
    try:
        path.find_command('ansible-playbook')
        if ansible_install_policy == 'python_install':
            LOG_JOB.debug("'Check if full ansible version is installed")
            cmd = r"pip freeze | grep -v ansible-core | grep ansible="
            process.run(cmd, verbose=False, shell=True)
    except (path.CmdNotFoundError, process.CmdError):
        if ansible_install_policy == 'distro_install':
            package_list.insert(0, 'ansible')
        else:
            if not policy_map[ansible_install_policy]():
                return False
```
The original reason for whole `try` block was to ensure the ansible is installed via checking `ansible-playbook` command. I think the approach, I showed above, follows the original approach.

Reply:
Yes, if we can ensure the full ansible version is installed, we don't need to check the callback plugins. And I checked, `ansible-core` is a part of `ansible`.

`ansible-playbook` is provided by `ansible-core` but not `ansible`, so besides checking this command, we have to check what's the ansible version indeed.

Reply:
@PaulYuuu Thanks! Yes, I came to the same conclusion, that's exactly my point :+1:  

I will check with @yanan-fu  and if we all agree on this, let's give it a try :slightly_smiling_face:

Reply:
I have discussed with @lkotek , I agree with this conclusion, let's do it, thanks!

Reply:
Great, thank you both!

Test using the lastest change passed (with `ansible-core` installed only):
```
JOB ID     : 5cee665ad4f1f9de095704a817a5d536374d5959
JOB LOG    : /root/avocado/job-results/job-2023-05-17T09.37-5cee665/job.log
 (1/1) L1_deployment.Host_RHEL.m9.u3.v0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.ansible_test: STARTED
 (1/1) L1_deployment.Host_RHEL.m9.u3.v0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.ansible_test: PASS (3663.94 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
```
Key part from the log:
```
[stdlog] 2023-05-17 09:37:23,895 avocado.test DEBUG| 'Check if full ansible version is installed
[stdlog] 2023-05-17 09:37:25,712 avocado.utils.process INFO | Running 'which pip'
[stdlog] 2023-05-17 09:37:25,733 avocado.utils.process DEBUG| [stdout] /usr/local/bin/pip
[stdlog] 2023-05-17 09:37:25,734 avocado.utils.process INFO | Command 'which pip' finished with 0 after 0.001849529s
[stdlog] 2023-05-17 09:37:25,735 avocado.utils.process INFO | Running 'pip install ansible'
[stdlog] 2023-05-17 09:37:27,544 avocado.utils.process DEBUG| [stdout] Collecting ansible
[stdlog] 2023-05-17 09:37:27,965 avocado.utils.process DEBUG| [stdout]   Using cached ansible-7.5.0-py3-none-any.whl (43.6 MB)
```

Reply:
Please see the latest test results here: https://github.com/autotest/tp-qemu/pull/3688#discussion_r1205533136